src/workspace.coffee

```diff
@@ -565,8 +565,9 @@ class Workspace extends Model
       @grammarRegistry, @project, @assert, @applicationDelegate
     }, params)
     editor = new TextEditor(params)
-    disposable = atom.textEditors.add(editor)
```

Comment:
I would rather just require the editor to be explicitly registered rather than having this extra parameter.